packages/firebase_auth/README.md

```diff
@@ -1,14 +1,126 @@
-# Firebase Authentication for Flutter
+# firebase_auth plugin
+A Flutter plugin to use the [Firebase Authentication API](https://firebase.google.com/products/auth/).
 
 [![pub package](https://img.shields.io/pub/v/firebase_auth.svg)](https://pub.dartlang.org/packages/firebase_auth)
 
-A Flutter plugin to use the [Firebase Authentication API](https://firebase.google.com/products/auth/).
-
 *Note*: This plugin is still under development, and some APIs might not be available yet. [Feedback](https://github.com/flutter/flutter/issues) and [Pull Requests](https://github.com/flutter/plugins/pulls) are most welcome!
 
 ## Usage
-To use this plugin, add `firebase_auth` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).
 
-## Getting Started
+### Import dependent package
+First install the dependency. Follow the [Google sign-in plugin installation instructions](https://pub.dartlang.org/packages/google_sign_in#pub-pkg-tab-installing).
+
+### Import this package
+To use this plugin, follow the [plugin installation instructions](https://pub.dartlang.org/packages/firebase_auth#pub-pkg-tab-installing).
+
+### Android integration
```

Comment:
I'd prefer having excellent docs on the google sign in plugin and then have a section about "Using firebase auth with google sign in" for this plugin that links to google sign in and makes it clear which steps from there need to be performed to make it work. In the future, there might be additional sections like that for all the other auth mechanisms that firebase_auth supports (but we currently don't support in the flutter plugin).